src/common/core/mixincore.js

```diff
@@ -0,0 +1,817 @@
+/*globals define*/
+/*jshint node: true, browser: true*/
+/**
+ * @author kecso / https://github.com/kecso
+ */
+
+define([
+    'common/util/assert',
+    'common/core/tasync',
+    'common/util/canon',
+    'common/core/constants'
+], function (ASSERT, TASYNC, CANON, CONSTANTS) {
+    'use strict';
+
+    var MixinCore = function (innerCore, options) {
+        ASSERT(typeof options === 'object');
+        ASSERT(typeof options.globConf === 'object');
+        ASSERT(typeof options.logger !== 'undefined');
+
+        var logger = options.logger,
+            key,
+            self = this;
+
+        for (key in innerCore) {
+            this[key] = innerCore[key];
+        }
+
+        logger.debug('initialized MixinCore');
+
+        //<editor-fold=Helper Functions>
+
+        function realNode(node) { //TODO we have to make some more sophisticated distinction
+            if (self.getPath(node).indexOf('_') !== -1) {
+                return false;
+            }
+            return true;
+        }
+
+        function getOrderedMixinList(node) {
+            var paths = self.getMixinPaths(node),
+                metaNodes = self.getAllMetaNodes(node),
```

Comment:
Not sure if getAllMetaNodes is expensive or not, but if paths is empty - there's no need to get it.